TobogganApp.ui/src/Components/Forms/ProductsForm.js

```diff
@@ -0,0 +1,167 @@
+          }, 3000);
+        });
+      } else {
+        const productObjectUpdate = {
+          Id: this.state.id,
+          Title: this.state.title,
+          Description: this.state.description,
+          Price: priceNum,
+          Quantity: quantityNum,
+          ShopId: shopIdNum,
+          CategoryId,
+          ProductImage: this.state.productImage,
+        };
+        productData.updateProduct(productObjectUpdate).then(() => {
+          this.setState({ success: true });
+          setTimeout(() => {
+            this.props.history.push('/user-dashboard/my-shop');
+          }, 3000);
+        });
+      }
+    };
+
+    render() {
+      const { success } = this.state;
+      return (
+            <div className="shopForm mr-auto ml-auto mt-5">
+        {success && (
+          <div className='alert alert-success' role='alert'>
+            Your Product was Updated/Created
+          </div>
+        )}
+        <form onSubmit={this.handleSubmit}>
+        {this.state.title}
+          <div>
+            <input
+              type='text'
+              name='title'
+              value={this.state.title}
+              onChange={this.handleChange}
+              placeholder='Product Name'
+              className='form-control form-control-lg m-2'
+              required
+            />
+          </div>
+          <div>
+            <input
+              type='text'
+              name='description'
+              value={this.state.description}
+              onChange={this.handleChange}
+              placeholder='Product Description'
+              className='form-control form-control-lg m-2'
+              required
+            />
+          </div>
+          <div>
+            <input
+              type='text'
+              name='price'
+              value={this.state.price}
+              onChange={this.handleChange}
+              placeholder='Product Price'
+              className='form-control form-control-lg m-2'
+              required
+            />
+          </div>
+          <div>
+            <input
+              type='text'
+              name='quantity'
+              value={this.state.quantity}
+              onChange={this.handleChange}
+              placeholder='Enter a Quanity'
+              className='form-control form-control-lg m-2'
+              required
+            />
+          </div>
+          <div>
+            <input
+              type='text'
+              name='productImage'
+              value={this.state.productImage}
+              onChange={this.handleChange}
+              placeholder='Enter an Image URL'
+              className='form-control form-control-lg m-2'
+              required
+            />
+          </div>
+          <div>
+              <select name='CategoryId' onChange={this.handleChange} >
+              {this.state.allCategories.map((category) => (<option value={category.id}>{category.name}</option>))}
```

Comment:
You're getting multiples of the same categories on here

Reply:
Mockaroo data!